bip-unknown-var-budget-script.mediawiki

```diff
@@ -0,0 +1,722 @@
+
+{|
+! Opcode
+! Value
+! Required Stack Elements
+! Definition
+|-
+|OP_2MUL
+|141
+|1
+|
+# Pop A off the stack.
+# Shift each byte in A 1 bit to the left (increasing values, equivalent to C's << operator), tracking the last non-zero value.
+# If the final byte overflows, append a single 1 byte.
+# Otherwise, truncate A at the last non-zero byte.
+# Push A onto the stack.
+|-
+|OP_2DIV
+|142
+|1
+|
+# Pop A off the stack.
+# Shift each byte 1 bit to the right (decreasing values, equivalent to C's >> operator), tracking the last non-zero value.
+# Truncate A at the last non-zero byte.
+# Push A onto the stack.
+|--
+|OP_MUL
+|149
+|2
+|
+# Pop B off the stack.
+# Pop A off the stack.
+# Calculate the varops cost of the operation: if it exceeds the remaining budget, fail.
+# Allocate an all-zero vector R of length equal to length(A) + length(B).
+# For each word in A, multiply it by B and add it into the vector R, offset by the word offset in A.
+# Truncate R at the last non-zero byte.
+# Push R onto the stack.
+|-
+|OP_DIV
+|150
+|2
+|
+# Pop B off the stack.
+# Pop A off the stack.
+# Calculate the varops cost of the operation: if it exceeds the remaining budget, fail.
+# If B is empty or all zeroes, fail.
+# Perform division as per Knuth's The Art of Computer Programming v2 page 272, Algorithm D "Division of non-negative integers".
+# Trim trailing zeroes off the quotient.
+# Push the quotient onto the stack.
+|-
+|OP_MOD
+|151
+|2
+|
+# Calculate the varops cost of the operation: if it exceeds the remaining budget, fail.
+# If B is empty or all zeroes, fail.
+# Perform division as per Knuth's The Art of Computer Programming v2 page 272, Algorithm D "Division of non-negative integers".
+# Trim trailing zeroes off the remainder.
+# Push the remainder onto the stack.
+|}
+
+
+===Hashing Opcodes===
+
+The SHA2-256 operation is already a key component of Bitcoin so an optimal implementation is assumed, and cost-per-byte-hashed defined below.
+
+OP_RIPEMD160 and OP_SHA1 are now defined to FAIL if their operands exceed 520 bytes.
+
+===Non-Arithmetic Opcodes Dealing With Stack Numbers===
+
+The following opcodes are redefined in v2 Tapscript to read numbers from the stack as arbitrary-length little-endian values (instead of CScriptNum):
+
+1. OP_CHECKLOCKTIMEVERIFY
+2. OP_CHECKSEQUENCEVERIFY
+3. OP_VERIFY
+4. OP_ROLL
+5. OP_IFDUP
+6. OP_CHECKSIGADD
+
+These opcodes are redefined in v2 Tapscript to write numbers to the stack as minimal-length little-endian values (instead of CScriptNum):
+
+1. OP_CHECKSIGADD
+2. OP_DEPTH
+3. OP_SIZE
+
+In addition, the success requirement in BIP-342 is modified to require a non-zero variable-length unsigned integer value (not <code>CastToBool()</code>):
+
+# If the execution results in anything but exactly one element on the stack which contains one or more non-zero bytes, fail.
```

Comment:
How about modelling this BIPs specification section after [BIP 342's specification section](https://github.com/bitcoin/bips/blob/master/bip-0342.mediawiki#specification) or mention the diff. This would make the change much more visible (and has other benefits in terms of clarity).